providers/common/provider_seeding.c

```diff
@@ -18,26 +18,30 @@ static OSSL_FUNC_cleanup_nonce_fn *c_cleanup_nonce = NULL;
 int ossl_prov_seeding_from_dispatch(const OSSL_DISPATCH *fns)
 {
     for (; fns->function_id != 0; fns++) {
+        /*
+        * We do not support the scenario of an application linked against
+        * multiple versions of libcrypto (e.g. one static and one dynamic), but
+        * sharing a single fips.so. We do a simple sanity check here.
+        */
+#define set_func(c, f) if (c != NULL && c != f) goto err; else c = f;
```

Comment:
Tiny little bit more efficient way is:
`#define set_func(c, f) if (c == NULL) c = f; else if (c != f) goto err;`